### ISSUE TITLE
Version 6 of #Angular is now available

### ISSUE THREAD

**Issue by DanielFran** [jhipster/generator-jhipster#7575]
Version 6 of #Angular is now available! Updates to the framework, CLI, Material, CDK, and more! 

https://blog.angular.io/version-6-of-angular-now-available-cc56b0efa7a4

https://blog.ninja-squad.com/2018/05/04/what-is-new-angular-6/

https://books.ninja-squad.com/angular/changelog

**Comment by sendilkumarn** [comment]
@wmarques go ahead. But a lot of `rxjs` refactorings will be there

**Comment by sendilkumarn** [comment]
with `ng-bootstrap-2.0.0` out. we can upgrade it

**Comment by agoncal** [comment]
@wmarques I'll be more than happy to test and give you some feedback.

BTW Angular CLI 6 makes it easy to create libraries (https://github.com/angular/angular-cli/wiki/stories-create-library). We use this "kind" of mecanism to aggregate our Microservices. Maybe something to brainstorm for the future

**Comment by Tcharl** [comment]
I noticed some jh issues to make angular.cli working with jhipster, for example, the manifest shall be in a .json extension instead of .manifest either way 'ng build' fails.

I'm also thinking of making a material module, as one of our company project's as done the refactoring work (but not on a module), but it's huge to implement during free time: may be if anyone could help, I can share our sources (under NDA) and so we could work on it together.

@wmarques  if you need help, I can spend some time on your fork.

@agoncal the library aspect is kind on a must have for us too: ng-packagr was not that easy to configure.

**Comment by wmarques** [comment]
@sendilkumarn Yep, in fact it's the rxjs part that is taking more time than the Angular parts... I'm submitting a PR with the beginning of my work, I don't think the build will pass for the first time 😢

**Comment by mraible** [comment]
This is the route that I took in Ionic for JHipster. Example: https://github.com/oktadeveloper/generator-jhipster-ionic/blob/master/generators/entity-client/templates/ionic/src/pages/entities/_entity-dialog.ts#L110 <https://github.com/oktadeveloper/generator-jhipster-ionic/blob/master/generators/entity-client/templates/ionic/src/pages/entities/_entity-dialog.ts#L110>

**Comment by thainan10** [comment]
@mraible sure, is with form builder. Angular calls it Reactive Forms.
I know we'll have a lot of effort to change all forms, but I think it worth.
What is the best approach in this case?
Should I create issues to change it and submit pull requests to solve them?

**Comment by wmarques** [comment]
@thainan10 It's not related with Angular 6 right ? But yes we can do that change, I also prefer the Reactive Form way...
You can submit a new issue or directly submit a PR if you already started working on it.

The thing that is TODO also is to migrate from having services in `providers` to use the new DI injection syntax in the `@Injectable` Annotation directly. But I think we should first merge my PR and then migrate the rest

**Comment by thainan10** [comment]
@wmarques sure. The Reactive Forms initiative is not related to the new Angular version, but I think important.
I can help you at first to refactor the services to the new model if necessary.
